lib/text/ttml_text_parser.js

```diff
@@ -202,38 +235,82 @@ shaka.text.TtmlTextParser = class {
    * @param {!Array.<!Element>} regionElements
    * @param {!Array.<!shaka.text.CueRegion>} cueRegions
    * @param {boolean} whitespaceTrim
+   * @param {boolean} isChild
    * @return {shaka.text.Cue}
    * @private
    */
   static parseCue_(
       cueElement, offset, rateInfo, metadataElements, styles, regionElements,
-      cueRegions, whitespaceTrim) {
+      cueRegions, whitespaceTrim, isChild) {
+    if (isChild && cueElement.nodeName === 'br') {
+      const cue = new shaka.text.Cue(0, 0, '');
+      cue.spacer = true;
+
+      return cue;
+    }
+
     // Disregard empty elements:
     // TTML allows for empty elements like <div></div>.
     // If cueElement has neither time attributes, nor
     // non-whitespace text, don't try to make a cue out of it.
-    if (!cueElement.hasAttribute('begin') &&
-        !cueElement.hasAttribute('end') &&
-        /^\s*$/.test(cueElement.textContent)) {
+    if (
+      /^[\s\n]*$/.test(cueElement.textContent) ||
+      (
+        !isChild &&
+        !cueElement.hasAttribute('begin') &&
+        !cueElement.hasAttribute('end')
+      )
+    ) {
       return null;
     }
 
-    shaka.text.TtmlTextParser.addNewLines_(cueElement, whitespaceTrim);
-
     // Get time.
     let start = shaka.text.TtmlTextParser.parseTime_(
         cueElement.getAttribute('begin'), rateInfo);
     let end = shaka.text.TtmlTextParser.parseTime_(
         cueElement.getAttribute('end'), rateInfo);
     const duration = shaka.text.TtmlTextParser.parseTime_(
         cueElement.getAttribute('dur'), rateInfo);
-    const payload = cueElement.textContent;
+    let payload = '';
+    const nestedCues = [];
+    // If one of the children is text node type
+    // stop going down and write the payload
+    if (
+      Array.from(cueElement.childNodes).find(
+          (childNode) => childNode.nodeType === Node.TEXT_NODE &&
+          /\w+/.test(childNode.textContent)
+      )
+    ) {
+      payload = shaka.text.TtmlTextParser.sanitizeTextContent(
+          cueElement,
+          whitespaceTrim,
+      );
+    } else {
+      for (let i = 0; i < cueElement.childNodes.length; i++) {
+        const childNode = cueElement.childNodes[i];
+        const childCue = shaka.text.TtmlTextParser.parseCue_(
+            /** @type {!Element} */ (childNode),
+            offset,
+            rateInfo,
+            metadataElements,
+            styles,
+            regionElements,
+            cueRegions,
+            whitespaceTrim,
+            true
```

Comment:
style nit: please add a comment with the parameter name: /* isChild */ true (assuming this is for isChild, which, I think, is the case)